script.js

```diff
@@ -1,4 +1,208 @@
+  return sum;
+}
+
+var determineWinner = function() {
+
+  if (primaryGameMode == NORMALMODE && player1Num > player2Num ||
+    primaryGameMode == LOWESTCOMBINEDNUMER && player1Num < player2Num) {
+    return 1;}
+    
+    return 2;
+  
+  };
+
+
+var leaderBoard =  function () {
+  if (player1TotalScore > player2TotalScore) {
+    return `Player 1 is in the Lead!<br><br>
+    Player 1 Total Score ${player1TotalScore}<br><br>
+    Player 2 Total Score ${player2TotalScore}`
+  } else
+   return `Player 2 is in the Lead!<br><br>
+   Player 2 Total Score ${player2TotalScore}<br><br>
+   Player 1 Total Score ${player1TotalScore}`
+}
+
+var lowestCombinedNumberBoard = function () {
+  if (player1TotalScore < player2TotalScore) {
+    return `Player 1 is in the Lead!<br><br>
+    Player 1 Total Score ${player1TotalScore}<br><br>
+    Player 2 Total Score ${player2TotalScore}`
+  } else
+   return `Player 2 is in the Lead!<br><br>
+   Player 2 Total Score ${player2TotalScore}<br><br>
+   Player 1 Total Score ${player1TotalScore}`
+}
+
+var main = function(input) {
+  
+  if (!primaryGameMode && !input) {
+    //user did not key in anything
+    return MODE_SELECTION_INSTRUCTIONS;
+  }
+
+  if (input == 1) {
+  primaryGameMode = NORMALMODE;}
+  else if (input == 2) {
+  primaryGameMode = LOWESTCOMBINEDNUMER;
+  };
+
+  if ((primarygameMode = NORMALMODE || LOWESTCOMBINEDNUMER) && gameMode == DICEROLL) {
+    var newDiceRolls = getDiceRolls();
+    gameMode = DICEORDER;
+    return `Welcome Player ${currPlayer}.<br><br>
+    You rolled Dice 1: ${newDiceRolls[0]} and Dice 2: ${newDiceRolls[1]}.
+    Please choose which dice you want to be first.` 
+}
+
+  if ((primarygameMode = NORMALMODE || LOWESTCOMBINEDNUMER) && gameMode == DICEORDER) {
+    var diceOrder = Number(input)
+    if (diceOrder !== 1 && diceOrder !== 2) {
+      return `Please choose 1 or 2.`
+    } else 
+
+    var playerNum = getPlayerNumber(diceOrder);
+
+  }
+
+    console.log (player1Dice)
+    console.log (player1Num)
+    
+    var playerMess = `Player ${currPlayer}, You chose Dice ${diceOrder} to be first.<br><br>
+    Your number is ${playerNum}.`
+
+    if (currPlayer == 1) {
+
+      player1SumNum.push(player1Num);
+
+      player1TotalScore = calcTotalNumber(player1SumNum);
+
+      currPlayer = 2;
+      gameMode = DICEROLL;
+
+      return `${playerMess}<br><br>
+      Player 2, Press Submit to roll dice`;
+     };
+
+    var whoWon = determineWinner();
+
+    player2SumNum.push(player2Num);
+
+    player2TotalScore = calcTotalNumber(player2SumNum);
+
+
+    currPlayer = 1;
+    gameMode = DICEROLL;
+
+    console.log(player1SumNum)
+    console.log(player2SumNum)
+    console.log(player1TotalScore)
+    console.log(player2TotalScore)
+     
+    if (primaryGameMode == NORMALMODE) {
+  
+    return `Player ${whoWon} wins!<br><br>
+    ${leaderBoard()} <br><br>
+    Press Submit To Play Again.`
+
+  }
+    
+    else  {
+      return `Player ${whoWon} wins!<br><br>
```

Comment:
Good use of template literals to help make the code more readable! Also makes it easier to type long strings  out, as opposed to concatenating them.  with "" and +